src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/config/ChangelogMigration.kt

```diff
@@ -0,0 +1,130 @@
+package uk.gov.justice.digital.hmpps.hmppsinterventionsservice.config
+
+import mu.KLogging
+import org.springframework.context.annotation.Configuration
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.dto.ReferralAmendmentDetails
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.Changelog
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.ReferralDetails
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.repository.AuthUserRepository
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.repository.ChangelogRepository
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.repository.ReferralDetailsRepository
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.service.AmendTopic
+import java.time.OffsetDateTime
+import java.time.temporal.ChronoUnit
+import java.util.UUID
+import javax.annotation.PostConstruct
+
+@Configuration
+class ChangelogMigration(
+  private val referralDetailsRepository: ReferralDetailsRepository,
+  private val changelogRepository: ChangelogRepository,
+  private val authUserRepository: AuthUserRepository
+) {
+  companion object : KLogging()
+
+  @PostConstruct
+  fun migrateDataFromReferralDetailsToChangeLog() {
+
+    logger.info("migration of referral details to change log starts")
+
+    // Get all the referral details where the superseded id not null
+
+    val referralDetails = referralDetailsRepository.findAll()
+
+    referralDetails.filter { it.supersededById != null }.forEach {
+      try {
+        val changeLogs = changelogRepository.findByReferralIdOrderByChangedAtDesc(it.referralId)
+        // Do this migration when there is no change log data
+        if (changeLogs.isEmpty()) {
```

Comment:
The changelog might contain entries from other amendments that happened; the emptiness check would miss those.